### PR TITLE
[YUNIKORN-2613] Release notes for 1.5.1

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Add release annoucement page for 1.5.1 

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [x] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2613

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: src/pages/release-announce/1.5.1.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+We are pleased to announce that the Apache YuniKorn community has voted to release 1.5.1. 
+Apache YuniKorn is a standalone resource scheduler, designed for managing and scheduling Batch and Data workloads on container
+orchestration frameworks like Kubernetes for on-prem and on-cloud use cases.
+
+## Overview
+The Apache YuniKorn community has resolved 18 [JIRAs](https://issues.apache.org/jira/issues/?filter=12353383) in this release.
+
+Release manager: Peter Bacsko
+
+Release date: 2024-05-15
```
Comment (wilfred-s) [inline src/pages/release-announce/1.5.1.md:35]: might need an update to 16th

File: src/pages/release-announce/1.5.1.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Highlights
+
+### Locking fixes to avoid existing and potential deadlocks
+
+YuniKorn 1.5.0 contained a deadlock which is now fixed in 1.5.1 along with other potential
+deadlocks. See [YUNIKORN-2553](https://issues.apache.org/jira/browse/YUNIKORN-2553) for details.
```
Comment (wilfred-s) [inline src/pages/release-announce/1.5.1.md:42]: Change to:

YuniKorn 1.5.0 contained a deadlock as described in [YUNIKORN-2521](https://issues.apache.org/jira/browse/YUNIKORN-2521). This deadlock is now fixed along with other potential deadlocks. See [YUNIKORN-2544](https://issues.apache.org/jira/browse/YUNIKORN-2544) for a more detailed overview of the locking changes made.

Comment (pbacsko) [reply src/pages/release-announce/1.5.1.md]: Done.

File: src/pages/release-announce/1.5.1.md (hunk 1 above)
Comment (pbacsko) [reply src/pages/release-announce/1.5.1.md]: Done

**Review by craigcondit — APPROVED** [review]
+1 LGTM. See minor nit, feel free to fix on merge.

**Pull request closed without merging**